src/runtime/components/forms/Input.vue

```diff
@@ -172,11 +177,40 @@ export default defineComponent({
       }
     }
 
-    const onInput = (event: InputEvent) => {
-      emit('update:modelValue', (event.target as HTMLInputElement).value)
+    // Custom function to handle the v-model properties
+    const updateInput = (value: string) => {
+
+      if (props.modelModifiers.trim) {
+        value = value.trim()
+      }
+
+      if (props.modelModifiers.number || props.type === 'number') {
+        value = looseToNumber(value)
+      }
+
+      emit('update:modelValue', value)
       emitFormInput()
     }
 
+    const onInput = (event: InputEvent) => {
+      if (!props.modelModifiers.lazy) {
+        updateInput((event.target as HTMLInputElement).value)
+      }
+    }
+
+    const onChange = (event: InputEvent) => {
+      const value = (event.target as HTMLInputElement).value
+
+      if (props.modelModifiers.lazy) {
+        updateInput(value)
+      }
+
+      // Update trimmed input so that it has same behaviour as native input https://github.com/vuejs/core/blob/5ea8a8a4fab4e19a71e123e4d27d051f5e927172/packages/runtime-dom/src/directives/vModel.ts#L63
+      if (props.modelModifiers.trim) {
+        (event.target as HTMLInputElement).value = value.trim()
+      }
```

Comment:
Isn't it already handled in `updateInput`?

Reply:
updateInput will only update the modelValue itself and not what is represented in the input field. If you add spaces after some text and tab out of the input it should trim what is in the field and not leave the spaces you added. Check the link i left from the v-model code.